core/dbt/tests/fixtures/project.py

```diff
@@ -404,11 +404,16 @@ def create_test_schema(self, schema_name=None):
 
     # Drop the unique test schema, usually called in test cleanup
     def drop_test_schema(self):
-        with get_connection(self.adapter):
-            for schema_name in self.created_schemas:
-                relation = self.adapter.Relation.create(database=self.database, schema=schema_name)
-                self.adapter.drop_schema(relation)
-            self.created_schemas = []
+        for schema_name in self.created_schemas:
+            relation = self.adapter.Relation.create(database=self.database, schema=schema_name)
+            schema = self.adapter.quote_as_configured(relation.schema, "schema")
+            self.run_sql(f"drop schema if exists {schema} cascade")
```

Comment:
I'd definitely prefer us _not_ going back to hard-coded `run_sql` here. This will definitely cause issues on other adapters.

Do we know _why_ we're running `drop schema` at a point in time when there are still open transactions against objects in the schema? In theory, every test cases uses its own dedicated schema, so by the time we're running `drop_test_schema`, we should be totally done with it

Reply:
@jtcohen6 from what I can tell it seems like a lock gets put on test_basic and Incremental_unique_id because they both reference the same tables between tests e.g `snapshot_expected`, and `duplicate_insert` and wonder if test hasn't had enough time to tear down ref to that table before next test kicks off especially if in same class so it locks as to not have two people updating same. table. as mentioned in https://docs.aws.amazon.com/redshift/latest/dg/r_STV_LOCKS.html.

Reply:
> the same tables

These should be in different schemas, though—a dedicated schema for each test case, for all the resources required for that test—right?

Reply:
Yes, every tests should get a randomly generated unique schema. So in theory different tests should not be locking each other's tables. And drop_test_schema should only be executed at the end of the test.